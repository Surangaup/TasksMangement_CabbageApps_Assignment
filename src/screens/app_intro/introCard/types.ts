export interface SlideProps {
    key:number,
    image: any,
    title: string,
    description: string
}