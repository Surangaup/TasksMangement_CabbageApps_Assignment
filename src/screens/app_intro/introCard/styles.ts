import { StyleSheet } from "react-native";
import { CommonStyles } from "../../../theme/styles";

export const styles = StyleSheet.create({ 
    container: { 
        justifyContent:'center',
        alignItems: 'center',
    }, 
    introContainer: {
        flex: 7, 
        backgroundColor: 'white',
    },
    loginContainer: {
        flex: 4
    },
    title: {
        fontSize: 20, color: 'black', 
        ...CommonStyles.baseFont_SemiBold
    },
    text: {
        color:'#9391A1',
        textAlign: 'center',
        padding: 10,
        ...CommonStyles.baseFont_Lite
    }
})