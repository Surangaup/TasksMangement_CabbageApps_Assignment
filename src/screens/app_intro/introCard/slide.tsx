import * as React from 'react';
import { Text, View, Image, ListRenderItemInfo } from 'react-native';
import { styles } from './styles';
import { SlideProps } from './types';

export const Slide = ({item}: ListRenderItemInfo<SlideProps>) => {
  console.log(item);
  
  return (
    <View style={styles.container} >
      <Image source={item.image}
        style={{ height: 240, width: 204 }}
        resizeMode='cover'
        resizeMethod='resize' />
      <Text style={styles.title}>{item.title}</Text>
      <Text style={styles.text}>{item.description}</Text>
    </View>
  );
}