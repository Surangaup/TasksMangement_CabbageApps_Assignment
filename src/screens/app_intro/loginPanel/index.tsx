import * as React from 'react';
import { View } from 'react-native';
import { LoginButtonProps } from './types';
import { styles } from './styles';
import Button from './button';
import { useLinkTo } from '@react-navigation/native';

const LoginButton = () => {
  const linkTo = useLinkTo();
  const [buttons] = React.useState<LoginButtonProps[]>([
    {
      image: require('../../../asserts/images/facebook.png'),
      title: 'Continue with Facebook',
      hasBackgroundColor: true
    },
    {
      image: require('../../../asserts/images/google.png'),
      title: 'Continue with Google',
      hasBackgroundColor: true
    },
    {
      image: require('../../../asserts/images/email.png'),
      title: 'Continue with Email',
      hasBackgroundColor: false
    }
  ])
  return (
    <View style={styles.container}>
      {
        buttons.map((button, index) =>
          <Button
            key={index}
            hasBackgroundColor={button.hasBackgroundColor}
            image={button.image}
            title={button.title}
            onPress={() => {
              linkTo('/Home')
            }}
          />
        )
      }
    </View>
  );
};

export default LoginButton;