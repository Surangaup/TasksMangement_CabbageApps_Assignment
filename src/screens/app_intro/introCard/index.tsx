import * as React from 'react';
import AppIntroSlider from 'react-native-app-intro-slider';
import { Slide } from './slide';
import { SlideProps } from './types';

interface AppIntroProps { }

export const IntroCards = (props: AppIntroProps) => {
  const [showRealApp, setShowRealApp] = React.useState<boolean>(false);

  const slides: SlideProps[] = [
    {
      key: 1,
      title: 'Organize your works',
      description: 'Let\'s organize your works with priority and do everything without stress.',
      image: require('../../../asserts/images/intro_1.png')
    },
    {
      key: 2,
      title: 'Title 2',
      description: '2nd Description.',
      image: require('../../../asserts/images/intro_1.png')
    }
  ];
  const _onDone = () => {
    setShowRealApp(true)
  }
  return (
      <AppIntroSlider
        renderItem={Slide}
        data={slides}
        onDone={_onDone}
        activeDotStyle={{ backgroundColor: '#FD92A1' }}
      />
  );
};