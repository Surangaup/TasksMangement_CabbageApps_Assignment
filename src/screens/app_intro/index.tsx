import * as React from 'react';
import { View } from 'react-native';
import {IntroCards} from './introCard';
import { styles } from './introCard/styles';
import LoginButtons from './loginPanel';

const AppIntro = () => {
    return (
        <React.Fragment>
            <View style={styles.introContainer}>
                <IntroCards />
            </View>
            <View style={styles.loginContainer}>
                <LoginButtons />
            </View>
        </React.Fragment>
    );
};

export default AppIntro;