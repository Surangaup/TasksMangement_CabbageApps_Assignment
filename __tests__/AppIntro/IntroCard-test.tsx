import * as React from 'react';
import renderer from 'react-test-renderer';
import {IntroCards} from '../../src/screens/app_intro/introCard';

test(('IntroCards snapshot'), ()=>{
    const snap = renderer.create(<IntroCards />).toJSON();
    expect(snap).toMatchSnapshot();
})