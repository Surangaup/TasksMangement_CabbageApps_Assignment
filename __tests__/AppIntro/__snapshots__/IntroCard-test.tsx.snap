// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroCards snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bottomButton={false}
    bounces={false}
    data={
      Array [
        Object {
          "description": "Let's organize your works with priority and do everything without stress.",
          "image": Object {
            "testUri": "../../../src/asserts/images/intro_1.png",
          },
          "key": 1,
          "title": "Organize your works",
        },
        Object {
          "description": "2nd Description.",
          "image": Object {
            "testUri": "../../../src/asserts/images/intro_1.png",
          },
          "key": 2,
          "title": "Title 2",
        },
      ]
    }
    disableVirtualization={false}
    dotClickEnabled={true}
    extraData={1}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={2}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showDoneButton={true}
    showNextButton={true}
    showPrevButton={false}
    showSkipButton={false}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMethod="resize"
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/asserts/images/intro_1.png",
                }
              }
              style={
                Object {
                  "height": 240,
                  "width": 204,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Gilroy SemiBold",
                  "fontSize": 20,
                }
              }
            >
              Organize your works
            </Text>
            <Text
              style={
                Object {
                  "color": "#9391A1",
                  "fontFamily": "Gilroy Lite",
                  "padding": 10,
                  "textAlign": "center",
                }
              }
            >
              Let's organize your works with priority and do everything without stress.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMethod="resize"
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/asserts/images/intro_1.png",
                }
              }
              style={
                Object {
                  "height": 240,
                  "width": 204,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Gilroy SemiBold",
                  "fontSize": 20,
                }
              }
            >
              Title 2
            </Text>
            <Text
              style={
                Object {
                  "color": "#9391A1",
                  "fontFamily": "Gilroy Lite",
                  "padding": 10,
                  "textAlign": "center",
                }
              }
            >
              2nd Description.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 16,
        "justifyContent": "center",
        "left": 16,
        "position": "absolute",
        "right": 16,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 16,
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FD92A1",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "padding": 12,
              }
            }
          >
            Next
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
